### PR TITLE
Improved event logic

### PR DESCRIPTION
**Pull request by andyguenin**

### CODE REVIEW

File: src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+class EventRequest(conn: Connection, settings: Settings) extends BaseRequest
+  with DatabaseRequest with UUIDProvider {
+
+  def getAllServiceEvents(serviceId: UUID): Seq[Event] = {
+    getAllEventsInternal(Some(serviceId), None)
+  }
+
+  def getEvent(serviceId: UUID, eventId: UUID): Event = {
+    getAllEventsInternal(Some(serviceId), Some(eventId)) match {
```
Comment (tzinckgraf) [inline src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala:21]: What if multiple get returned? Is that not possible?

File: src/test/scala/org/downtowndailybread/bethsaida/model/ScheduleDetailTest.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import java.util.TimeZone
+
+class ScheduleDetailTest extends FlatSpec {
+
+  implicit val time = LocalDateTime.of(2020, 1, 5, 5, 0, 0)
+    .atZone(TimeZone.getDefault.toZoneId)
+  val startTime = LocalTime.of(11, 0, 0)
+  val endTime = LocalTime.of(13, 0, 0)
+
+  "a scheudle that includes the RRULE: prefix" should "work" in {
```
Comment (tzinckgraf) [inline src/test/scala/org/downtowndailybread/bethsaida/model/ScheduleDetailTest.scala]: Boo. Spelling.

File: src/main/scala/org/downtowndailybread/bethsaida/worker/EventScheduler.scala [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import scala.concurrent.duration.Duration
+
+class EventScheduler(settings: Settings) extends Actor {
+
+  override def preStart(): Unit = {
+    self ! CheckUpcomingEvents
+  }
+
+
+  override def receive: Receive = {
```
Comment (tzinckgraf) [inline src/main/scala/org/downtowndailybread/bethsaida/worker/EventScheduler.scala:24]: Where does this get called? It's definitely something I'll need on an endpoint.

File: src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala (hunk 1 above)
Comment (andyguenin) [reply src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala:21]: It should throw an event not found exception. Cause it doesn’t know which one to return.

Comment (andyguenin) [reply src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala:21]: It should throw an event not found exception. Cause it doesn’t know which one to return.

File: src/main/scala/org/downtowndailybread/bethsaida/worker/EventScheduler.scala (hunk 3 above)
Comment (andyguenin) [reply src/main/scala/org/downtowndailybread/bethsaida/worker/EventScheduler.scala:24]: Future PR. It’s a background activity called outside the web endpoint.

File: src/test/scala/org/downtowndailybread/bethsaida/model/ScheduleDetailTest.scala (hunk 2 above)
Comment (andyguenin) [reply src/test/scala/org/downtowndailybread/bethsaida/model/ScheduleDetailTest.scala]: I must have ignored IntelliJ

File: src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+      s"""
+         |insert into event (id, service_id, metadata_id)
+         |values (cast(? as uuid), cast(? as uuid), ?)
+       """.stripMargin
+
+    val ps = conn.prepareStatement(sql)
+    ps.setString(1, eventId)
+    ps.setString(2, serviceId)
+    ps.setInt(3, insertMetadataStatement(conn, true))
+    insertEventAttributeInternal(eventId, event, true)
```
Comment (tzinckgraf) [inline src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala:39]: Do both of these functions insert into event, or am I going crazy?

File: src/main/scala/org/downtowndailybread/bethsaida/request/util/DatabaseRequest.scala [hunk 5]
```diff
@@ -14,6 +18,30 @@ trait DatabaseRequest {
...
+    }
+    def setNullableString(parameterIndex: Int, x: Option[String]): Unit = {
+      x match {
+        case Some(i) => ps.setString(parameterIndex, i)
+        case None => ps.setNull(parameterIndex, java.sql.Types.VARCHAR)
+      }
+    }
+    def setNullableUUID(parameterIndex: Int, x: Option[UUID]): Unit = {
+      x match {
+        case Some(i) => ps.setString(parameterIndex, i)
```
Comment (tzinckgraf) [inline src/main/scala/org/downtowndailybread/bethsaida/request/util/DatabaseRequest.scala:36]: Does this automatically convert the UUID to a string?

File: src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala (hunk 4 above)
Comment (andyguenin) [reply src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala:39]: it's a two part process: the master record which does the relational bit, and then the event_attribute info which has the actual data in it.

File: src/main/scala/org/downtowndailybread/bethsaida/request/util/DatabaseRequest.scala (hunk 5 above)
Comment (andyguenin) [reply src/main/scala/org/downtowndailybread/bethsaida/request/util/DatabaseRequest.scala:36]: ya, UUIDProvider on line 11:

```
implicit def uuidToString(uuid: UUID): String = uuid.toString
```

File: src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    createSeq(ps.executeQuery(), eventCreator)
+  }
+
+  private def insertEventAttributeInternal(eventId: UUID, event: EventAttribute, isValid: Boolean)(
+    implicit iu: InternalUser
+  ): Unit = {
+    val metaId = insertMetadataStatement(conn, true)
+    val sql =
+      s"""
+         |insert into event
```
Comment (tzinckgraf) [inline src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala]: Given your above comment, should this be `event_attribute`?

Comment (andyguenin) [reply src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala]: fixed

**Pull request merged**